src/AstGen.zig

```diff
@@ -1722,6 +1722,55 @@ fn structInitExpr(
         }
     }
 
+    // First: The first found field.
+    // Duplicates: A list of following duplicate field names.
+    const DupeNameRecord = struct { first: Ast.TokenIndex, duplicates: ArrayListUnmanaged(Ast.TokenIndex) };
+
+    var duplicate_names = std.AutoHashMap(u32, DupeNameRecord).init(astgen.arena);
+    try duplicate_names.ensureTotalCapacity(@intCast(struct_init.ast.fields.len));
+
+    // // Basic check to maintain O(N) best case.
```

Comment:
This comment is a little odd, because, well, it'd be O(n) either way. It's fairly clear what this is being used for, so the comment isn't really necessary, but if you do want to keep it around, correct it:
```suggestion
    // When there aren't errors, use this to avoid a second iteration.
```